cmake_test/add_test.cmake

```diff
@@ -1,5 +1,20 @@
 include_guard()
-include(cmake_test/detail_/add_test)
+include(cmake_test/detail_/utilities/print_result)
+
+function(dump_cmake_variables)
+    get_cmake_property(_variableNames VARIABLES)
```

Comment:
please namespace prefix variables to avoid collisions (underscores are rarely enough).